RFCs/FS-1129-shorthand-anonymous-unary-functions.md

```diff
@@ -0,0 +1,208 @@
+
+* Debugging
+  * Breakpoints/stepping
+    * This should behave in the same way as the equivalent elaborated form: a breakpoint covering a line with the shorthand should break at least once each time the function is called. Similarly, stepping into the function should behave as if it had been written in the elaborated form.
+  * Expression evaluator
+    * TODO
+  * Data displays for locals and hover tips
+    * The debugger should show the value of the synthetic local variable storing the argument when stopped inside the function call
+    * When inside the scope of multiple nested versions of this shorthand the variables should have unique (and somehow clear) names
+    * The current synthetic variable value should be displayed when hovering over the underscore part of the shorthand.
+      * Not any underscore, and the correct value for the correct function when nested
+    * Other data displays and hover should be equivalent to the elaborated form
+* Auto-complete
+  * Completion after `_.` should be equivalent to completion after `(fun x -> x.` in the same context.
+  * TODO: Check there aren't any snippets or other oddities
+* Tooltips
+  * On hovering over the underscore a tooltip should be shown the the type of the anonymous function, and any other context that would normally be shown for a function arg.
+  * TODO: Does the rhs `x` ever show something different to the lhs `x`?
+* Navigation and Go To Definition
+  * None? No document symbols are created by this construct.
+* Colorization
+  * TODO
+* Brace/parenthesis matching
+  * TODO
+
+## Performance
+
+TODO, but I can't see how this could cause any impact on the performance of generated code as the construct should be entirely erased.
+
+This adds rules to the parser which could cause more backtracking.
+
+This adds extra work between parsing and checking which requires walking the syntax tree.
+
+## Scaling
+
+Please list the dimensions that describe the inputs for this new feature, e.g. "number of widgets" etc.  For each, estimate a reasonable upper bound for the expected size in human-written code and machine-generated code that the compiler will accept.
+
+For example
+
+* Expected maximum number of widgets in reasonable hand-written code: 100
+* Expected reasonable upper bound for number of widgets accepted: 500
+
+Testing should particularly check that compilation is linear (or log-linear or similar) along these dimensions.  If quadratic or worse this should ideally be noted in the RFC.
+
+## Culture-aware formatting/parsing
+
+TODO, but no
+
+Does the proposed RFC interact with culture-aware formatting and parsing of numbers, dates and currencies? For example, if the RFC includes plaintext outputs, are these outputs specified to be culture-invariant or current-culture.
+
+# Unresolved questions
+
+1. Is underscore definitely the appropriate symbol?
+2. Is there a good name for this feature?
+3. What happens when you nest this?
```

Comment:
At the moment, as specified, no significant confusing with regard to nesting appears possible.